nifi-extension-bundles/nifi-gitlab-bundle/nifi-gitlab-extensions/src/main/java/org/apache/nifi/gitlab/GitLabFlowRegistryClient.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.registry.flow.FlowRegistryClientConfigurationContext;
+import org.apache.nifi.registry.flow.FlowRegistryException;
+import org.apache.nifi.registry.flow.git.AbstractGitFlowRegistryClient;
+import org.apache.nifi.registry.flow.git.client.GitRepositoryClient;
+import org.gitlab4j.api.GitLabApi;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Tags({"git", "gitlab", "registry", "flow"})
+@CapabilityDescription("Flow Registry Client that uses the GitLab REST API to version control flows in a GitLab Project.")
+public class GitLabFlowRegistryClient extends AbstractGitFlowRegistryClient {
+
+    static final PropertyDescriptor GITLAB_API_URL = new PropertyDescriptor.Builder()
+            .name("GitLab API URL")
+            .description("The URL of the GitLab API")
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .defaultValue("https://gitlab.com/")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor GITLAB_API_VERSION = new PropertyDescriptor.Builder()
+            .name("GitLab API Version")
+            .description("The version of the GitLab API")
+            .allowableValues(GitLabApi.ApiVersion.class)
+            .defaultValue(GitLabApi.ApiVersion.V4.name())
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor REPOSITORY_NAMESPACE = new PropertyDescriptor.Builder()
+            .name("Repository Namespace")
+            .description("The namespace of the repository. Typically the name of a group that owns the repository.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor REPOSITORY_NAME = new PropertyDescriptor.Builder()
+            .name("Repository Name")
+            .description("The name of the repository")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor AUTHENTICATION_TYPE = new PropertyDescriptor.Builder()
+            .name("Authentication Type")
+            .description("The type of authentication to use for accessing GitLan")
+            .allowableValues(GitLabAuthenticationType.class)
+            .defaultValue(GitLabAuthenticationType.ACCESS_TOKEN.name())
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor ACCESS_TOKEN = new PropertyDescriptor.Builder()
+            .name("Access Token")
+            .description("The access token to use for authentication")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .required(true)
+            .sensitive(true)
+            .dependsOn(AUTHENTICATION_TYPE, GitLabAuthenticationType.ACCESS_TOKEN.name())
```

Comment:
If `GitLabAuthenticationType` was a `DescribedValue` this could be simplified minimally to:
```suggestion
            .dependsOn(AUTHENTICATION_TYPE, GitLabAuthenticationType.ACCESS_TOKEN)
```